File: 11.PrefixSum前缀和/leetcode_2574_leftRightDifference.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> leftRightDifference(vector<int>& nums) {
        int n = nums.size();
        vector<int> lSum(n, 0), rSum(n, 0);
        for (int i=0; i<n-1; i++){
            lSum[i+1] = lSum[i]+nums[i];
            rSum[n-i-2] = rSum[n-i-1]+nums[n-i-1];
        }
        vector<int> res;
        for (int i=0; i<n; i++) res.push_back(abs(lSum[i]-rSum[i]));
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {10,4,8,3};
    vector<int> res = s.leftRightDifference(nums);
    for (int i : res){
        cout << i << " ";
    }
    cout << endl;
}

File: 11.PrefixSum前缀和/leetcode_2574_leftRightDifference.py
from typing import List

class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        n = len(nums)
        lSum = [0]*n
        rSum = [0]*n
        for i in range(n-1):
            lSum[i+1] = lSum[i]+nums[i]
            rSum[n-i-2] = rSum[n-i-1]+nums[n-i-1]
        return [abs(lSum[i]-rSum[i]) for i in range(n)]

if __name__ == "__main__":
    s = Solution()
    nums = [10,4,8,3]
    print(s.leftRightDifference(nums))
